### DISCUSSION TITLE
Welcome to gRASPA Discussions!

### DISCUSSION THREAD

**Discussion by Zhaoli2042** [snurr-group/gRASPA#17] [Announcements]
## 👋 Welcome to the Discussion channel of gRASPA!
  You can:
  * Ask questions about the code.
  * Share ideas about using the code.
  * Query about new features and future updates.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by inangoi** [comment]
Hi, 
Where in gRASPA I can find the amount of gas adsorbed?  
In RASPA2 it is written as: 

--------------------------------------
Component 0 [H2O]
-------------------------------------------------------------
        Block[ 0] 0.05944            [-]
        Block[ 1] 0.05654            [-]
        Block[ 2] 0.05622            [-]
        Block[ 3] 0.05712            [-]
        Block[ 4] 0.05625            [-]
        ------------------------------------------------------------------------------
        Average loading absolute                              0.0571160000 +/-       0.0016767927 [-]
        Average loading absolute [molecules/unit cell]        0.0071395000 +/-       0.0002095991 [-]
        Average loading absolute [mol/kg framework]                  0.0026140428 +/-       0.0000767422 [-]
        Average loading absolute [milligram/gram framework]          0.0470919816 +/-       0.0013825109 [-]
        Average loading absolute [cm^3 (STP)/gr framework]           0.0585910927 +/-       0.0017200980 [-]
        Average loading absolute [cm^3 (STP)/cm^3 framework]         0.0541725008 +/-       0.0015903784 [-]
------------------------------------------

When I run the same structure in gRASPA I have the following information, that is quite different from RASPA2 :
---------------------------------------
COMPONENT [1] (H2O)
Framework component 0, molar mass: 2728.56000
Framework total mass: 21828.48000
BLOCK [0], Blocksize: 600000, Average: 26.18167, ErrorBar: 0.45710
BLOCK [1], Blocksize: 600000, Average: 26.87801, ErrorBar: 0.57880
BLOCK [2], Blocksize: 600000, Average: 27.51076, ErrorBar: 0.21306
BLOCK [3], Blocksize: 600000, Average: 27.83755, ErrorBar: 0.16868
BLOCK [4], Blocksize: 600000, Average: 27.98495, ErrorBar: 0.22682
Overall: Average: 27.27859, ErrorBar: 1.33536
--------------------------------

Thank you in advance if someone could help me.

PhD. Inna M. Nangoi
UFJF - GFQSI


I will put the inputs below: 
gRASPA:
UseGPUReduction no
Useflag yes

NumberOfInitializationCycles 1000000
NumberOfEquilibrationCycles  0
NumberOfProductionCycles     3000000

UseMaxStep  yes
MaxStepPerCycle 1

UseChargesFromCIFFile yes

RestartFile no
RandomSeed  0

BMCBiasingMethod LJ_Biasing

NumberOfTrialPositions 10
NumberOfTrialOrientations 10
NumberOfBlocks 1
AdsorbateAllocateSpace 10240
NumberOfSimulations 1
SingleSimulation yes

DifferentFrameworks yes
InputFileType cif
FrameworkName ZIF-8-resp
UnitCells 0 2 2 2

ChargeMethod Ewald
Temperature 298.0
Pressure    100000

OverlapCriteria 1e5
CutOffVDW 14.0
CutOffCoulomb 14.0
EwaldPrecision 1e-6

UseDNNforHostGuest no

Component 0 MoleculeName             H2O
            IdealGasRosenbluthWeight 1.0
            FugacityCoefficient      1.0
            TranslationProbability   0.5
            RotationProbability      0.5
            ReinsertionProbability   0.5
            SwapProbability          1.0
            CreateNumberOfMolecules  0


RASPA2:

SimulationType                MonteCarlo
NumberOfCycles                3000000
NumberOfInitializationCycles  1000000
PrintEvery                    1000
RestartFile                   no

Forcefield                    ExampleMOFsForceField
UseChargesFromCIFFile         yes

Framework                     0
FrameworkName                 ZIF-8-resp
UnitCells                     2 2 2
HeliumVoidFraction            0.48
ExternalTemperature           298.0
ExternalPressure              100000

Component 0 MoleculeName             H2O
            MoleculeDefinition       ExampleDefinitions
            TranslationProbability   0.5
            RotationProbability      0.5
            ReinsertionProbability   0.5
            SwapProbability          1.0
            CreateNumberOfMolecules  0

**Reply by inangoi** [reply]
Below are the gRASPA data (completed)


=====================BLOCK AVERAGES (LOADING: mol/kg)=============
COMPONENT [0] (ZIF-8-resp.cif)
Framework component 0, molar mass: 2728.56000
Framework total mass: 21828.48000
BLOCK [0], Blocksize: 600000, Average: 0.04581, ErrorBar: 0.00000
BLOCK [1], Blocksize: 600000, Average: 0.04581, ErrorBar: 0.00000
BLOCK [2], Blocksize: 600000, Average: 0.04581, ErrorBar: 0.00000
BLOCK [3], Blocksize: 600000, Average: 0.04581, ErrorBar: 0.00000
BLOCK [4], Blocksize: 600000, Average: 0.04581, ErrorBar: 0.00000
Overall: Average: 0.04581, ErrorBar: 0.00000
----------------------------------------------------------
COMPONENT [1] (H2O)
Framework component 0, molar mass: 2728.56000
Framework total mass: 21828.48000
BLOCK [0], Blocksize: 600000, Average: 26.18167, ErrorBar: 0.45710
BLOCK [1], Blocksize: 600000, Average: 26.87801, ErrorBar: 0.57880
BLOCK [2], Blocksize: 600000, Average: 27.51076, ErrorBar: 0.21306
BLOCK [3], Blocksize: 600000, Average: 27.83755, ErrorBar: 0.16868
BLOCK [4], Blocksize: 600000, Average: 27.98495, ErrorBar: 0.22682
Overall: Average: 27.27859, ErrorBar: 1.33536

**Reply by Zhaoli2042** [reply]
Hi @inangoi 

Thanks for using the code. My email is zhaoli2023@u.northwestern.edu. If you don't mind, could you please email me the input files and output files of gRASPA and RASPA-2 (if there is nothing sensitive) and then I can inspect the files and tell you what is different, and we can further discuss the issue. 

Thanks,
Zhao

**Comment by inangoi** [comment]
Hi Zhao, 

Can gRASPA calculate RDF ? What are the flags? 

Thank you in advance,

Inna

**Reply by Zhaoli2042** [reply]
Hi, 

No, for now, it cannot do RDF. Hope this helps.

Best,
Zhao

**Comment by Zackary-Overend** [comment]
Hi Zhao

Just a quick question but how do you switch on the flag for CFC (if this has been implimented)? I know that the following is for CBMC

NumberOfTrialPositions 10
NumberOfTrialOrientations 10

But I would like to run CFC or CBCFC if possible.

Many thanks
Zack Overend
PhD Student, Sarkisov Group
